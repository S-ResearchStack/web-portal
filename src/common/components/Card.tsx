import React, { ForwardedRef, forwardRef, isValidElement, ReactElement } from 'react';

import styled, { css, keyframes } from 'styled-components';

import RefreshAnimatedIcon from 'src/assets/icons/refresh_animated.svg';
import { boxShadow, colors, px, typography } from 'src/styles';
import ServiceScreen from 'src/common/components/ServiceScreen';

export const BASE_CARD_PADDING = 24;

type BaseCardProps = React.PropsWithChildren<Omit<React.HTMLAttributes<HTMLDivElement>, 'title'>>;

export interface CardProps extends BaseCardProps, React.RefAttributes<HTMLDivElement> {
  title?: string | React.ReactNode;
  subtitle?: string | boolean | React.ReactNode;
  action?: React.ReactNode;
  loading?: boolean;
  empty?: boolean;
  error?: boolean;
  contentChanging?: boolean;
  onReload?: () => void;
  bottomAction?: boolean;
  height?: string;
}

type CardContainerProps = React.PropsWithChildren<Pick<CardProps, 'loading'>>;

const CardContainer = styled.div.withConfig({
  shouldForwardProp: (prop, defaultValidatorFn) =>
    defaultValidatorFn(prop) && !['loading'].includes(prop),
})<CardContainerProps>`
  display: flex;
  flex-direction: column;
  background-color: ${colors.surface};
  box-shadow: ${boxShadow.card};
  border-radius: ${px(4)};
  box-sizing: border-box;
  padding: ${px(BASE_CARD_PADDING)} ${px(BASE_CARD_PADDING)} ${px(16)};
  overflow: visible;
  position: relative;
  ${({ loading }) =>
    loading &&
    css`
      pointer-events: none;
      * {
        cursor: default;
      }
    `}

  > * {
    min-width: 0;
    min-height: 0;
    overflow: visible;
  }
`;

type ContentProps = React.PropsWithChildren<Pick<CardProps, 'loading' | 'contentChanging'>>;

const Content = styled(({ loading, contentChanging, ...props }: ContentProps) => (
  <div {...props} />
))`
  transition: ${({ loading, contentChanging, theme }) =>
    !loading && !contentChanging && `opacity 0.3s ${theme.animation.defaultTiming}`};
  opacity: ${({ loading, contentChanging }) => (loading || contentChanging ? 0 : 1)};
  flex: 1;
  position: relative;
  display: flex;
  flex-direction: column;
`;

const AppearanceDelayAnim = keyframes`
  0% { opacity: 0; }
  80% { opacity: 0; }
  100% { opacity: 1; }
`;

export const TitleContainer = styled.div<{ hasSubtitle?: boolean }>`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const Title = styled.div<{ hasSubtitle: boolean; bottomAction?: boolean }>`
  display: flex;
  justify-content: space-between;
  height: ${({ hasSubtitle }) => (hasSubtitle ? px(21) : px(43))};

  ${({ bottomAction }) =>
    bottomAction &&
    css`
      flex-direction: column;
      row-gap: ${px(5)};
    `};
`;

const TitleText = styled.p`
  ${typography.headingXSmall};
  color: ${colors.textPrimaryDark};
  display: inline-block;
  margin: 0;
`;

const Subtitle = styled.div<{ subtitle?: string | boolean }>`
  ${typography.bodySmallRegular};
  color: ${colors.textSecondaryGray};
  min-height: ${({ subtitle }) => (subtitle === true ? px(20) : px(43))};
  min-width: ${px(50)};
  display: flex;
  margin-top: ${px(8)};
`;

export const RefreshIconStyled = styled(RefreshAnimatedIcon)`
  margin-left: ${px(9)};
  flex-shrink: 0;
  animation: ${AppearanceDelayAnim} 1.25s;
`;

export const Action = styled(({ loading, ...props }) => <div {...props} />)`
  display: ${({ loading }) => (loading ? 'none' : 'flex')};
  overflow: visible;
  position: relative;
`;

const Card = forwardRef(
  (
    {
      title,
      subtitle,
      action,
      loading,
      empty,
      children,
      error,
      onReload,
      contentChanging,
      bottomAction,
      height,
      ...props
    }: CardProps,
    ref: ForwardedRef<HTMLDivElement>
  ): ReactElement  => {
    const renderContent = () => {
      if (error) {
        return <ServiceScreen onReload={onReload} type="error" data-testid="error-screen" />;
      }
      if (empty) {
        return <ServiceScreen type="empty" data-testid="empty-screen" />;
      }

      return children;
    };

    return (
      <CardContainer ref={ref} loading={loading} style={{height: height || 'fit-content'}} {...props}>
        {title && (
          <TitleContainer hasSubtitle={!!subtitle}>
            {isValidElement(title) ? (
              title
            ) : (
              <Title hasSubtitle={!!subtitle} bottomAction={bottomAction}>
                <TitleText data-testid="title">{title}</TitleText>
                {loading && <RefreshIconStyled data-testid="loader" />}
                {action && !error && <Action loading={loading}>{action}</Action>}
              </Title>
            )}
            {subtitle &&
              (isValidElement(subtitle) ? (
                subtitle
              ) : (
                <Subtitle data-testid="subtitle" subtitle={subtitle as string | boolean}>
                  {subtitle}
                </Subtitle>
              ))}
          </TitleContainer>
        )}
        <Content data-testid="card-content" loading={loading} contentChanging={contentChanging}>
          {renderContent()}
        </Content>
      </CardContainer>
    );
  }
);

export default Card;
