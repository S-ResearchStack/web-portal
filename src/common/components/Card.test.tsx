import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { getByText, render } from '@testing-library/react';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import Card from './Card';

describe('Card', () => {
  it('test card render', () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card data-testid="card" title="title" subtitle="subtitle">
          <div>Samsung</div>
        </Card>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');

    expect(card).toBeInTheDocument();
    expect(getByText(baseElement, 'title')).toBeInTheDocument();
    expect(getByText(baseElement, 'subtitle')).toBeInTheDocument();
    expect(getByText(baseElement, 'Samsung')).toBeInTheDocument();
  });

  it('[NEGATIVE] test card error state', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card data-testid="card" title="title" subtitle="subtitle" error>
          <div data-testid="content">Samsung</div>
        </Card>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');
    const content = queryByTestId('content');
    const errorScreen = queryByTestId('error-screen');

    expect(card).toBeInTheDocument();
    expect(getByText(baseElement, 'title')).toBeInTheDocument();
    expect(getByText(baseElement, 'subtitle')).toBeInTheDocument();
    expect(content).not.toBeInTheDocument();
    expect(errorScreen).toBeInTheDocument();
  });

  it('[NEGATIVE] test card empty state', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card data-testid="card" title="title" subtitle="subtitle" empty>
          <div data-testid="content">Samsung</div>
        </Card>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');
    const content = queryByTestId('content');
    const emptyScreen = queryByTestId('empty-screen');

    expect(card).toBeInTheDocument();
    expect(getByText(baseElement, 'title')).toBeInTheDocument();
    expect(getByText(baseElement, 'subtitle')).toBeInTheDocument();
    expect(content).not.toBeInTheDocument();
    expect(emptyScreen).toBeInTheDocument();
  });

  it('test card loading state', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card data-testid="card" title="title" subtitle="subtitle" loading>
          <div>Samsung</div>
        </Card>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');
    const content = queryByTestId('content');
    const title = queryByTestId('title');
    const subtitle = queryByTestId('subtitle');
    const loader = queryByTestId('loader');

    expect(card).toBeInTheDocument();
    expect(title).toBeInTheDocument();
    expect(subtitle).toBeInTheDocument();
    expect(content).not.toBeInTheDocument();
    expect(loader).toBeInTheDocument();
  });

  it('[NEGATIVE] should render wrong properties', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card
          data-testid="card"
          title={true as unknown as string}
          subtitle={true as unknown as string}
        >
          <div>{true}</div>
        </Card>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');
    const content = queryByTestId('card-content');
    const title = queryByTestId('title');
    const subtitle = queryByTestId('subtitle');

    expect(card).toBeInTheDocument();
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent('');
    expect(subtitle).toBeInTheDocument();
    expect(subtitle).toHaveTextContent('');
    expect(content).toBeInTheDocument();
    expect(subtitle).toHaveTextContent('');
  });

  it('[NEGATIVE] should render without props', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Card data-testid="card" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const card = getByTestId('card');
    const content = queryByTestId('card-content');
    const title = queryByTestId('title');
    const subtitle = queryByTestId('subtitle');

    expect(card).toBeInTheDocument();
    expect(title).toBeNull();
    expect(subtitle).toBeNull();
    expect(content).toBeInTheDocument();
  });
});
