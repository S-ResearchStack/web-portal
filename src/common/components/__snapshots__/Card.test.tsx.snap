// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card [NEGATIVE] should render without props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c1 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
        data-testid="card-content"
      />
    </div>
  </div>
</body>
`;

exports[`Card [NEGATIVE] should render wrong properties 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c5 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 1.3125rem;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 1.25rem;
  min-width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
            data-testid="title"
          />
        </div>
        <div
          class="c4"
          data-testid="subtitle"
        />
      </div>
      <div
        class="c5"
        data-testid="card-content"
      >
        <div />
      </div>
    </div>
  </div>
</body>
`;

exports[`Card [NEGATIVE] test card empty state 1`] = `
.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 100%;
  min-height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  margin-bottom: 0;
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-bottom: 1.4375rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c5 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 1.3125rem;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 2.6875rem;
  min-width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
            data-testid="title"
          >
            title
          </p>
        </div>
        <div
          class="c4"
          data-testid="subtitle"
        >
          subtitle
        </div>
      </div>
      <div
        class="c5"
        data-testid="card-content"
      >
        <div
          class="c6"
          data-testid="empty-screen"
        >
          <div
            class="c7"
          >
            <svg
              data-testid="empty-icon"
            />
          </div>
          <div
            class="c8"
          >
            No Data
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Card [NEGATIVE] test card error state 1`] = `
.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 100%;
  min-height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #2D2D2D;
  margin-bottom: 1.4375rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c5 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 1.3125rem;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 2.6875rem;
  min-width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
            data-testid="title"
          >
            title
          </p>
        </div>
        <div
          class="c4"
          data-testid="subtitle"
        >
          subtitle
        </div>
      </div>
      <div
        class="c5"
        data-testid="card-content"
      >
        <div
          class="c6"
          data-testid="error-screen"
        >
          <div
            class="c7"
          >
            Server Error
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Card test card loading state 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
  pointer-events: none;
}

.c0 * {
  cursor: default;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c6 {
  opacity: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 1.3125rem;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c5 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 2.6875rem;
  min-width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

.c4 {
  margin-left: 0.5625rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-animation: AGEkE 1.25s;
  animation: AGEkE 1.25s;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
            data-testid="title"
          >
            title
          </p>
          <svg
            class="c4"
            data-testid="loader"
          />
        </div>
        <div
          class="c5"
          data-testid="subtitle"
        >
          subtitle
        </div>
      </div>
      <div
        class="c6"
        data-testid="card-content"
      >
        <div>
          Samsung
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Card test card render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c5 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 1.3125rem;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 2.6875rem;
  min-width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="card"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
            data-testid="title"
          >
            title
          </p>
        </div>
        <div
          class="c4"
          data-testid="subtitle"
        >
          subtitle
        </div>
      </div>
      <div
        class="c5"
        data-testid="card-content"
      >
        <div>
          Samsung
        </div>
      </div>
    </div>
  </div>
</body>
`;
